### PR TITLE
fix - IAM User takeover Correlation Rule correlating on IP instead of user

### PR DESCRIPTION
**Pull request by akozlovets098**
### Background

Fixes https://github.com/panther-labs/panther-analysis/issues/1359

### Changes

- Made IAM User takeover Correlation Rule correlate on user instead of IP

### CODE REVIEW

**Review by arielkr256 — CHANGES_REQUESTED** [review]
I think this need to correlate on both IP and target user name.  We want to detect when Alice@11.22.33.44 resets Bob's password, then Bob logs in from 11.22.33.44.

Username Bob is in the `requestParameters.userName` field in the [PasswordReset](https://github.com/panther-labs/panther-analysis/blob/release/rules/aws_cloudtrail_rules/aws_cloudtrail_loginprofilecreatedormodified.py#L19) rule, and would be the last part of `userIdentity.arn` in the Login rule.

**Comment by akozlovets098** [comment]
> I think this need to correlate on both IP and target user name.

@arielkr256 We [cannot](https://docs.panther.com/detections/correlation-rules#:~:text=Only%20one%20type%20of%20field%20may%20be%20matched%20on%20throughout%20the%20correlation%20rule.%20For%20example%2C%20only%20IP%20addresses%20can%20be%20matched%20on%2C%20or%20only%20email%20addresses.) correlate on both. We can add some context field to both rules that will contain both IP and user name and try correlating on this list, but I'm not sure that this is a good idea (it is not an explicit approach) and that it will work. What do you think about that?

File: rules/aws_cloudtrail_rules/aws_console_login.py [hunk 1]
```diff
@@ -1,2 +1,10 @@
 def rule(event):
     return event.get("eventName") == "ConsoleLogin"
+
+
+def alert_context(event):
+    context = {}
+    context["ip_and_username"] = event.get(
+        "sourceIPAddress", "<MISSING_SOURCE_IP>"
+    ) + event.deep_get("userIdentity", "userName", default="<MISSING_USER_NAME>")
```
Comment (arielkr256) [inline rules/aws_cloudtrail_rules/aws_console_login.py:9]: Cloudtrail logs don't have a `userIdentity.userName` field.  You'll need to parse the name from after the last `/` in `userIdentity.arn`

arn:aws:sts::98765432:assumed-role/IAM/**alice**

**Review by arielkr256 — APPROVED** [review]
LGTM!

**Pull request merged**